subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -772,25 +780,27 @@ const struct bt_mesh_model_cb _bt_mesh_scene_srv_cb = {
 static int scene_setup_srv_init(struct bt_mesh_model *model)
 {
 	struct bt_mesh_scene_srv *srv = model->user_data;
+	struct bt_mesh_dtt_srv *dtt_srv;
+	int err;
 
 	if (!srv) {
 		return -EINVAL;
 	}
 
 	srv->setup_mod = model;
 
-	/* Model extensions:
-	 * To simplify the model extension tree, we're flipping the
-	 * relationship between the scene server and the scene
-	 * setup server. In the specification, the scene setup
-	 * server extends the scene server, which is the opposite of
-	 * what we're doing here. This makes no difference for the mesh
-	 * stack, but it makes it a lot easier to extend this model, as
-	 * we won't have to support multiple extenders.
-	 */
-	bt_mesh_model_extend(srv->model, srv->setup_mod);
+	dtt_srv = bt_mesh_dtt_srv_get(bt_mesh_model_elem(model));
+	if (!dtt_srv || !dtt_srv->model) {
+		BT_ERR("Generic DTT Server must be initialized before Scene Server");
```

Comment:
Better to move this to setup or use the model find function, if possible, IMO. DTT may be part of a lightness server, and it seems reasonable that users would like to put the scene at the start of the element, then put models like lightness after it.